src/Sylius/Bundle/ApiBundle/DataTransformer/LoggedInCustomerAwareContactCommandDataTransformer.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\DataTransformer;
+
+use Sylius\Bundle\ApiBundle\Command\ContactEmailAwareInterface;
+use Sylius\Bundle\ApiBundle\Command\CustomerEmailAwareInterface;
+use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Sylius\Component\User\Model\UserInterface;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+final class LoggedInCustomerAwareContactCommandDataTransformer implements CommandDataTransformerInterface
+{
+    public function __construct(private UserContextInterface $userContext)
+    {
+    }
+
+    public function transform($object, string $to, array $context = [])
+    {
+        /** @var CustomerInterface|null $customer */
+        $customer = $this->getCustomer();
+
+        /** @var CustomerEmailAwareInterface|mixed $object */
+        Assert::object($object, ContactEmailAwareInterface::class);
+
+        if ($customer === null) {
+            return $object;
+        }
+
+        if (!$object->hasEmailSet()) {
```

Comment:
I don't understand why we need this function? can't we simply get an email from the object?

Reply:
IMO, if there is logged in customer, we should use an email of this customer even if there will be an email provided to the request

Reply:
@GSadee but it will be different behaviour than UI, where logged customer have own email in form but he can change it to a different email. I spoke about it also with @lchrusciel